src/realm/dictionary.cpp

```diff
@@ -336,7 +334,7 @@ bool Dictionary::is_null(size_t ndx) const
 
 Mixed Dictionary::get_any(size_t ndx) const
 {
-    update_if_needed();
+    // Note: `size()` calls `update_if_needed()`.
```

Comment:
@jedelbo Isn't that a quite unexpected side-effect judging from the name? Should it perhaps be renamed?

Reply:
@bmunkholm I am not sure I understand this comment. It should not be unexpected that `size()` calls `update_if_needed`.

Reply:
It just looked like that since the callsites calling size() called update_if_needed() before calling size() :-)